src/System.Security.AccessControl/src/System/Security/AccessControl/Rules.cs

```diff
@@ -0,0 +1,432 @@
+
+        #endregion
+    }
+
+
+    public abstract class ObjectAuditRule: AuditRule
+    {
+        #region Private Members
+
+        private readonly Guid _objectType;
+        private readonly Guid _inheritedObjectType;
+        private readonly ObjectAceFlags _objectFlags = ObjectAceFlags.None;
+
+        #endregion
+
+        #region Constructors
+
+        protected ObjectAuditRule( IdentityReference identity, int accessMask, bool isInherited, InheritanceFlags inheritanceFlags, PropagationFlags propagationFlags, Guid objectType, Guid inheritedObjectType, AuditFlags auditFlags )
+            : base( identity, accessMask, isInherited, inheritanceFlags, propagationFlags, auditFlags )
+        {
+            
+            if (( !objectType.Equals( Guid.Empty )) && (( accessMask & ObjectAce.AccessMaskWithObjectType ) != 0 ))
+            {
+                _objectType = objectType;
+                _objectFlags |= ObjectAceFlags.ObjectAceTypePresent;
+            }
+            else
+            {
+                _objectType = Guid.Empty;
+            }
+
+            if (( !inheritedObjectType.Equals( Guid.Empty )) && ((inheritanceFlags & InheritanceFlags.ContainerInherit ) != 0 ))
+            {
+                _inheritedObjectType = inheritedObjectType;
+                _objectFlags |= ObjectAceFlags.InheritedObjectAceTypePresent;
+            }
+            else
+            {
+                _inheritedObjectType = Guid.Empty;
+            }
+        }
+
+        #endregion
+
+        #region Public Properties
+
+        public Guid ObjectType
+        {
+            get { return _objectType; }
+        }
+
+        public Guid InheritedObjectType
+        {
+            get { return _inheritedObjectType; }
+        }
+
+        public ObjectAceFlags ObjectFlags
+        {
+            get { return _objectFlags; }
+        }
+
+
+        #endregion
+    }
+
+
+    public sealed class AuthorizationRuleCollection : ICollection, IEnumerable // TODO: Is this right? Was previously ReadOnlyCollectionBase
+    {
+        #region ReadOnlyCollectionBase APIs
+        // Goo to translate this from ReadOnlyCollectionBase to ICollection
+        Object _syncRoot;
+        List<AuthorizationRule> list;
+
+        List<AuthorizationRule> InnerList
+        {
+            get
+            {
+                if (list == null)
+                    list = new List<AuthorizationRule>();
+                return list;
+            }
+        }
+
+        public int Count
+        {
+            get { return InnerList.Count; }
+        }
+
+        bool ICollection.IsSynchronized
+        {
+            get { return false; }
+        }
+
+        object ICollection.SyncRoot
+        {
+            get
+            {
+                if (_syncRoot == null)
+                {
+                    System.Threading.Interlocked.CompareExchange<Object>(ref _syncRoot, new Object(), null);
+                }
+                return _syncRoot;
+            }
```

Comment:
This can just use `List<T>`'s `ICollection.SyncRoot` implementation, allowing the `_syncRoot` field to be removed, e.g. `return ((ICollection)InnerList).SyncRoot;`